addons/account/models/account_move.py

```diff
@@ -76,6 +76,8 @@ def _get_default_invoice_incoterm(self):
 
     # ==== Business fields ====
     name = fields.Char(string='Number', required=True, readonly=True, copy=False, default='/')
+    rate = fields.Float(digits=(12, 6), help='The between this currency and company currency')
+    has_foreign_currency = fields.Boolean(compute='_compute_has_foreign_currency')
```

Comment:
This boolean is not necessary, and in any case this should be store=True to use it in reports.

The account move with empty currency is company currency, any other is foreign.

Reply:
The computed boolean is there to hide/show the rate field. That was the only purpose, that's why is not stored.

Reply:
Ok, on that case, I think that you can do a `attrs={'invisible': [('company_id.currency_id.id', '!=', currency_id.id)]}` don't you think? and not new field required at all.